docs/docs/cloud/how-tos/stream_messages.md

```diff
@@ -0,0 +1,412 @@
+# How to stream messages from your graph
+
+LangGraph Cloud supports multiple streaming modes. The main ones are:
+
+- `values`: This streaming mode streams back values of the graph. This is the **full state of the graph** after each node is called.
+- `updates`: This streaming mode streams back updates to the graph. This is the **update to the state of the graph** after each node is called.
+- `messages`: This streaming mode streams back messages - both complete messages (at the end of a node) as well as **tokens** for any messages generated inside a node. This mode is primarily meant for powering chat applications.
+
+
+This guide covers `stream_mode="messages"`.
+
+In order to use this mode, the state of the graph you are interacting with MUST have a messages key that is a list of messages.
+Eg, the state should look something like:
+
+=== "Python"
+    ```python
+    from typing import TypedDict, Annotated
+    from langgraph.graph import add_messages
+    from langchain_core.messages import AnyMessage
+
+    class State(TypedDict):
+        messages: Annotated[list[AnyMessage], add_messages]
+    ```
+
+=== "Javascript"
+
+    ```js
+    import { AnyMessage } from "@langchain/core/messages";
+    import { StateGraphArgs } from "@langchain/langgraph";
+
+
+    // Define the state interface
+    interface AgentState {
+        messages: AnyMessage[];
+    }
+
+    // Define the graph state
+    const graphState: StateGraphArgs<AgentState>["channels"] = {
+        messages: {
+            value: (x: AnyMessage[], y: AnyMessage[]) => x.concat(y),
+            default: () => [],
+        },
+    };
+    ```
+
+OR it should be an instance or subclass of `from langgraph.graph import MessageState` (`MessageState` is just a helper type hint equivalent to the above).
```

Comment:
(nit) "it should be an instance or subclass of" -> "you can use [`MessageState`](link to api ref)".

Reply:
we should still add a link to api ref here